### DISCUSSION TITLE
How to implement unique field in collection?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [appwrite/appwrite#1492] [Q&A] [answered]
For an example, If i need to create master of parts and in which part number should be unique and shouldn't be duplicated how to enforce it?

In parseserver there is an event trigger option in which I used to validate whether unique before saving record and throw error. Whether appwrite have similar functionalities

**Comment by eldadfux** [comment]
This is not available at the moment, but will be with the release of the database refactor we have in progress.

![image](https://user-images.githubusercontent.com/1297371/130325858-28887d99-4afc-4077-bf54-760ca3c6de46.png)

**Comment by stnguyen90** [comment, marked as answer]
This is now available as of [0.12.0](https://github.com/appwrite/appwrite/releases/tag/0.12.0). For more info, please refer to our [docs](https://appwrite.io/docs/databases#indexes).
